corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js

```diff
@@ -85,10 +85,26 @@ if (typeof define === 'undefined') {
     define = hqDefine;
 }
 
+// For use only with modules that are never used in a requirejs context.
 function hqImport(path) {
     if (COMMCAREHQ_MODULES[path] === undefined) {
         throw new Error("The module '" + path + "' has not yet been defined.\n\n" +
             'Did you include <script src="' + path + '.js"></script> on your html page?');
     }
     return COMMCAREHQ_MODULES[path];
 }
+
+// Support require calls within a module. Best practice is to require all dependencies
+// at module definition time, but this function can be used when doing so would
+// introduce a circular dependency.
+function hqRequire(paths, callback) {
+    if (typeof define === 'function' && define.amd && window.USE_REQUIREJS) {
+        requirejs(paths, callback);
+    } else {
+        var args = [];
+        for (var i = 0; i < paths.length; i++) {
+            args.push(hqImport(paths[i]));
+        }
+        callback.apply(undefined, args);
```

Comment:
i think i instinctually prefer this is applied to an empty object rather than `window`, but i guess best would just be to mirror whatever requirejs does

Reply:
That's a good point, and I share your instinct (hadn't been thinking about this defaulting to window when I wrote this code). But it seems requirejs does default to window, so I'm going to leave the code as it is.